### PR TITLE
Updated and restructured Ch16.5, 16.6

### PR DESCRIPTION
**Pull request by Xmasreturns**
Readability and grammar changes
Renamed "Result as an alias" to "aliases for Result"
Made "map for Result" and "aliases for Result" into children of "Result"
Changed name of aliased result in Ch16.6 for clarity

### CODE REVIEW

**Comment by mdinger** [comment]
Haven't said it in a while but I'd like to reiterate that the work you're putting into improving the readability and comprehension of these examples is really great and appreciated. Thanks so much!

File: examples/structure.json [hunk 1]
```diff
@@ -143,9 +143,10 @@
         { "id": "unwrap", "title": "Option & unwrap", "children": null },
         { "id": "map", "title": "map", "children": null },
         { "id": "and_then", "title": "and_then", "children": null },
-        { "id": "result", "title": "Result", "children": null },
-        { "id": "result_map", "title": "map for Result", "children": null },
-        { "id": "result_alias", "title": "Result as an alias", "children": null },
+        { "id": "result", "title": "Result", "children": [
+            { "id": "result_map", "title": "map for Result", "children": null },
+            { "id": "result_alias", "title": "aliases for Result", "children": null },
```
Comment (mdinger) [inline examples/structure.json]: This trailing comma is probably breaking the build. The last element of a child can't have a trailing comma.

Comment (Xmasreturns) [inline examples/structure.json]: Shoot, I should've caught this one too :[

File: examples/error/result/result_alias/alias.rs [hunk 2]
```diff
@@ -0,0 +1,23 @@
+use std::num::ParseIntError;
+use std::result;
+
+// Define a generic alias for a `Result` of type `ParseIntError`.
```
Comment (steveklabnik) [inline examples/error/result/result_alias/alias.rs]: what about "with the error type ParseIntError"?

File: examples/error/result/result_alias/input.md [hunk 3]
```diff
@@ -0,0 +1,18 @@
+How about when we want to reuse a specific `Result` type many times? 
+It quickly becomes tedious to write out the full type name, but recall that Rust allows 
+us to create [aliases][alias]. A generic alias may conveniently be defined for the
```
Comment (steveklabnik) [inline examples/error/result/result_alias/input.md]: I might say "type alias" inside the link, since that's the full name of the feature.

**Comment by steveklabnik** [comment]
A few very very tiny nits, but looks good! Thanks so much :)

File: examples/error/result/result_alias/alias.rs (hunk 2 above)
Comment (Xmasreturns) [inline examples/error/result/result_alias/alias.rs]: More informative, I like it.

File: examples/error/result/result_alias/input.md (hunk 3 above)
Comment (Xmasreturns) [inline examples/error/result/result_alias/input.md]: Changed!

**Pull request merged**